BRB/run.py

```diff
@@ -24,49 +55,82 @@
+            )
+            print(f"Logging into: {logFile}")
+            setLog(logFile)
 
-        # Open log file
-        logFile = Path(
-            config['Paths']['logPath'],
-            config.get('Options','runID') + '.log'
-        )
-        print(f"Logging into: {logFile}")
-        setLog(logFile)
+        else:
+            # Push stats on-demand
+            log.info(f"Pushing stats for flowcell: {fcid}")
+            d = [d for d in glob.glob("{}/*/fastq.made".format(config.get('Paths', 'baseData'))) if fcid in d]
+            dual_lane = len(d) == 2
+            if len(d) == 0:
+                log.error(f"No fastq.made files found for {fcid}")
+                return  # Exit BRB if no files found.
+            elif len(d) > 2:
+                log.error(f"How many lanes does {fcid} have?!")
+                return  # Exit BRB this error shouldn't happen at all.
+            
+            config.set('Options','runID',d[0].split("/")[-2])
+            ParkourDict = BRB.findFinishedFlowCells.queryParkour(config)
+            
+            if dual_lane:
+                config1 = BRB.getConfig.getConfig(configfile)
+                config1.set('Options','runID',d[1].split("/")[-2])
+                ParkourDict1 = BRB.findFinishedFlowCells.queryParkour(config)
 
-        #Process each group's data, ignore cases where the project isn't in the lanes being processed
-        bdir = "{}/{}".format(config.get('Paths', 'baseData'), config.get('Options', 'runID'))
-        msg = []
-        for k, v in ParkourDict.items():
-            if not os.path.exists("{}/Project_{}".format(bdir, BRB.misc.pacifier(k))):
-                log.info("{}/Project_{} doesn't exist, probably lives on another lane.".format(bdir, BRB.misc.pacifier(k)))
-                continue
-            try:
-                msg = msg + BRB.PushButton.GetResults(config, k, v)
-            except Exception as e:
-                BRB.email.errorEmail(config, sys.exc_info(), "Received an error running PushButton.GetResults() with {} and {}".format(k, v))
-                log.critical("Received an error running PushButton.GetResults() with {} and {}".format(k, v))
-                print("Received an error running PushButton.GetResults() with {} and {}".format(k, v), file=sys.stderr)
-                raise
+            # Open log file
+            if not dual_lane:
+                logFile = Path(config['Paths']['logPath'], config.get('Options','runID') + '.log')
+            else:
+                logFile = Path(config['Paths']['logPath'], config.get('Options','runID') + '_2' + '.log')
+            print(f"Logging into: {logFile}")
+            setLog(logFile)
+
+            if dual_lane:
+                log.info("Same log-file is being used for both lanes. Hopefully this is not too confusing :$")
+
+        
+        # Process each group's data, ignore cases where the project isn't in the lanes being processed
+        process_data(config, ParkourDict)
```

Comment:
I still fail to understand the logic here, sorry..

stats not set -> only set a log file
stats set -> infer some lane status and set some other paths ?

Afterwards all the data gets processed anyway. For me this is effectively the same as running BRB twice ( with the added disadvantage that we are hardcoded for two lanes now).

If you don't have FID as an argument, but an actual path to a processed flowcell (which might be split per lane or not, but at least you don't have to infer it here), and subsequently have:
if not stats:
  process_data
else:
  if analysis was actually done already (because analysis worked, parkour was just dead) -> run phoneHome
  else no analysis was actually done (because non-std libtype, or external data) -> run telegraphHome